### DISCUSSION TITLE
Missing items when widening VirtuosoGrid

### DISCUSSION THREAD

**Discussion by harry0000** [petyosi/react-virtuoso#835] [Q&A] [answered]
Hi,

I wanted to use VirtuosoGrid in my application, so I tried it in CodeSandbox.

https://virtuoso.dev/grid-responsive-columns/#responsive-layout

There was one problem: If the style `display: grid` was set for ListContainer, items were missing when the grid was gradually widened from a narrow width.

<details>
<summary>screenshot</summary>

![image](https://user-images.githubusercontent.com/3132512/215303332-0ea95923-347c-47e5-aa60-8ba81452dfaf.png)

</details>

It seems to me that this is caused by the [gridState items](https://github.com/petyosi/react-virtuoso/blob/v4.0.6/src/gridSystem.ts#L139-L146) not updating when the grid gets wider, is there something wrong with my code? Or perhaps this is a problem that will be solved when the `increaseViewportBy` option is added to  VirtuosoGrid( #807 )?

<details>
<summary>App.js</summary>

```jsx
const ItemContainer = styled.div`
  width: 100px;
  height: 100px;
  display: flex;
`

const ItemWrapper = styled.div`
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 80%;
  border: 1px solid #000;
  white-space: nowrap;
`

const ListContainer = styled.div`
  display: grid;
  grid-gap: 10px;
  grid-template-columns: repeat(auto-fill, 100px);
  grid-template-rows: repeat(auto-fill, 100px);
  justify-content: space-evenly;
  margin: 10px;
`

export default function App() {
  return (
  <>
    <VirtuosoGrid
      style={{ height: 340 }}
      totalCount={10000}
      overscan={200}
      components={{
        Item: ItemContainer,
        List: ListContainer,
        ScrollSeekPlaceholder: ({ height, width, index }) => (
          <ItemContainer>
            <ItemWrapper>{'--'}</ItemWrapper>
          </ItemContainer>
        ),
      }}
      itemContent={index => <ItemWrapper>Item {index}</ItemWrapper>}
      scrollSeekConfiguration={{
        enter: velocity => Math.abs(velocity) > 200,
        exit: velocity => Math.abs(velocity) < 30,
        change: (_, range) => console.log({ range }),
      }}
    />
    <style>{`html, body, #root { margin: 0; padding: 0 }`}</style>
  </>
  )
}
```

</details>

https://codesandbox.io/s/sandpack-project-forked-uuz7rn

**Comment by petyosi** [comment, marked as answer]
This seems like a legit problem, not related to the `display: grid`, [here's a repro with flex](https://codesandbox.io/s/sandpack-project-forked-ywet1k?file=/App.js).

**Comment by nghiepdev** [comment]
I also have this issue with `display: grid`, but it only appears occasionally and randomly.

**Comment by petyosi** [comment]
Should be fixed by this: https://github.com/petyosi/react-virtuoso/issues/837. If you experience something, follow the issue template with a reproduction.

**Reply by nghiepdev** [reply]
😢 I'm using the latest version `^4.1.0`. I have issues with infinite scroll when using with `endReached={() => fetchNextPage()}`

**Reply by petyosi** [reply]
> Follow the issue template with a reproduction.
